[Question] How to cancel previous toast using toast method from anko common?

I want to cancel previous toast using `toast(message)` from anko commons. Previously in java i could make 1 static toast object and re-using it to auto cancel previous message.
Like in this java helper class that i made:
```java
public class CustomToast {
    private static Toast toast;

    public static void show(Context context, final String message) {
        final Context applicationContext = context.getApplicationContext();//make sure that there is no activity leak
        try {
            //make sure that the code will run on ui thread using handler post
            new Handler().post(
                    new Runnable() {
                        @Override
                        public void run() {
                            if (toast != null) {
                                toast.cancel();
                            }
                            toast = Toast.makeText(applicationContext, message, Toast.LENGTH_LONG);
                            toast.show();
                        }
                    }
            );
        } catch (Exception e) {
            //in case context is not present, etc
        }
    }
}
```

Is there any easy way to do this with anko?

Actually, I am trying to explore Kotlin and still a newbie in this field. I already cloned the project, had an error when building the project, either with android studio directly or with the steps from https://github.com/Kotlin/anko/wiki/Building-Anko
```
FAILURE: Build completed with 4 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':ide:intentions:compileKotlin'.
> Internal compiler error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':testUtils:compileKotlin'.
> Internal compiler error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':static:commons:compileReleaseKotlin'.
> Internal compiler error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':ide:attrs:compileKotlin'.
> Internal compiler error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

BUILD FAILED
```
I have search the github issues, but found nothing.
If you have any idea how to solve the problem, i would love to contribute back to this repository.

--- a/anko/library/static/commons/src/dialogs/Toasts.kt
+++ b/anko/library/static/commons/src/dialogs/Toasts.kt
@@ -40,7 +40,11 @@ inline fun Fragment.toast(message: Int) = activity.toast(message)
  *
  * @param message the message text resource.
  */
-fun Context.toast(message: Int) = Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
+inline fun Context.toast(message: Int): Toast = Toast
+        .makeText(this, message, Toast.LENGTH_SHORT)
+        .apply {
+            show()
+        }
 
 /**
  * Display the simple Toast message with the [Toast.LENGTH_SHORT] duration.
@@ -61,7 +65,11 @@ inline fun Fragment.toast(message: CharSequence) = activity.toast(message)
  *
  * @param message the message text.
  */
-fun Context.toast(message: CharSequence) = Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
+inline fun Context.toast(message: CharSequence): Toast = Toast
+        .makeText(this, message, Toast.LENGTH_SHORT)
+        .apply {
+            show()
+        }
 
 /**
  * Display the simple Toast message with the [Toast.LENGTH_LONG] duration.
@@ -82,7 +90,11 @@ inline fun Fragment.longToast(message: Int) = activity.longToast(message)
  *
  * @param message the message text resource.
  */
-inline fun Context.longToast(message: Int) = Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+inline fun Context.longToast(message: Int): Toast = Toast
+        .makeText(this, message, Toast.LENGTH_LONG)
+        .apply {
+            show()
+        }
 
 /**
  * Display the simple Toast message with the [Toast.LENGTH_LONG] duration.
@@ -103,4 +115,8 @@ inline fun Fragment.longToast(message: CharSequence) = activity.longToast(messag
  *
  * @param message the message text.
  */
-inline fun Context.longToast(message: CharSequence) = Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+inline fun Context.longToast(message: CharSequence): Toast = Toast
+        .makeText(this, message, Toast.LENGTH_LONG)
+        .apply {
+            show()
+        }

--- a/anko/library/static/supportV4/src/SupportDialogs.kt
+++ b/anko/library/static/supportV4/src/SupportDialogs.kt
@@ -22,13 +22,13 @@ import android.content.DialogInterface
 import android.support.v4.app.Fragment
 import org.jetbrains.anko.*
 
-inline fun Fragment.toast(textResource: Int): Unit = activity.toast(textResource)
+inline fun Fragment.toast(textResource: Int) = activity.toast(textResource)
 
-inline fun Fragment.toast(text: CharSequence): Unit = activity.toast(text)
+inline fun Fragment.toast(text: CharSequence) = activity.toast(text)
 
-inline fun Fragment.longToast(textResource: Int): Unit = activity.longToast(textResource)
+inline fun Fragment.longToast(textResource: Int) = activity.longToast(textResource)
 
-inline fun Fragment.longToast(text: CharSequence): Unit = activity.longToast(text)
+inline fun Fragment.longToast(text: CharSequence) = activity.longToast(text)
 
 inline fun Fragment.selector(
         title: CharSequence? = null,

